command/role.go

```diff
@@ -31,16 +30,43 @@ func role(ctx *Context, args []string) {
 		return
 	}
 
-	number, err := strconv.Atoi(args[1])
-	if err != nil {
-		ctx.ReportError("Invalid number", err)
-		return
-	} else if number < 0 || number >= len(ctx.Env.RoleColors) {
-		ctx.Reply("Invalid number")
-		return
+	type colorRole struct {
+		ID   string `json:"id"`
+		Name string `json:"color"`
+	}
+	var colorRoles []colorRole
+	for i, _ := range ctx.Env.RoleColors {
+		color, _ := ctx.Session.State.Role(ctx.Message.GuildID, ctx.Env.RoleColors[i])
+		name := fmt.Sprintf("%s", (color.Name))
```

Comment:
Wouldn't `name := color.Name` or something similar work here?

Reply:
ah my bad I was actually using `%s\n` but for some reason I removed the new line but forgot to remove `Sprintf`